p2p/transport/webtransport/transport_test.go

```diff
@@ -473,17 +473,29 @@ func TestAcceptQueueFilledUp(t *testing.T) {
 		return cl.Dial(context.Background(), ln.Multiaddr(), serverID)
 	}
 
-	for i := 0; i < 16; i++ {
-		conn, err := newConn()
-		require.NoError(t, err)
-		defer conn.Close()
+	const num = 16 + 1 // one more than the accept queue capacity
+	// Dial one more connection than the accept queue can hold.
+	errChan := make(chan error, num)
+	for i := 0; i < num; i++ {
+		go func() {
+			conn, err := newConn()
+			if err != nil {
+				errChan <- err
+				return
+			}
+			_, err = conn.AcceptStream()
+			errChan <- err
+		}()
 	}
 
-	conn, err := newConn()
-	if err == nil {
-		_, err = conn.AcceptStream()
-	}
-	require.Error(t, err)
+	// Since the handshakes complete asynchronously, we won't know _which_ one is rejected,
+	// so the only thing we can test for is that exactly one connection attempt is rejected.
+	require.Eventually(t, func() bool { return len(errChan) > 0 }, time.Second, 10*time.Millisecond, "expected an error to occur")
```

Comment:
Can we recv from this channel with a timeout instead of polling?

Reply:
Makes sense.